### PR TITLE
[EVM-528]: Resolve TODOs in state code

### PR DESCRIPTION
**Pull request by goran-ethernal**
# Description

Resolved TODOs in `state` package. Some TODOs are obsolete and removed.

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the relevant labels
- [ ] I have updated the official documentation
- [ ] I have added sufficient documentation in code

## Testing

- [x] I have tested this code with the official test suite
- [x] I have tested this code manually

### CODE REVIEW

**Review by vcastellm — APPROVED** [review]

File: state/immutable-trie/trie.go [hunk 1]
```diff
@@ -113,12 +113,10 @@ func hashit(k []byte) []byte {
...
 
-var accountArenaPool fastrlp.ArenaPool
-
-// TODO, Remove once we do update in fastrlp (to be fixed in EVM-528)
-//
-//nolint:godox
-var stateArenaPool fastrlp.ArenaPool
+var (
+	accountArenaPool fastrlp.ArenaPool
+	stateArenaPool   fastrlp.ArenaPool
```
Comment (Stefan-Ethernal) [inline state/immutable-trie/trie.go]: Could we merge these two? I think that was the esence of the TODO which is deleted.

File: state/txn.go [hunk 2]
```diff
@@ -492,9 +492,6 @@ func (txn *Txn) TouchAccount(addr types.Address) {
 	})
 }
 
-//nolint:godox
```
Comment (Stefan-Ethernal) [inline state/txn.go:495]: Could we do something regarding panics? For example 2 out of the 3 panics are thrown in the `CleanDeleteObjects`. Couldn't we add error propagation there instead?

File: state/immutable-trie/trie.go (hunk 1 above)
Comment (goran-ethernal) [reply state/immutable-trie/trie.go]: I would rather not risk it. Not worth it.

Comment (Stefan-Ethernal) [reply state/immutable-trie/trie.go]: I would, no risk no gain 😉
Let me try it out. Arena is just a some kind of helper strict which acts as builder if I'm not mistaken, so think it should be safe changing it.

**Comment by Stefan-Ethernal** [comment]
IBFT E2E tests triggered manually:
- https://github.com/0xPolygon/polygon-edge/pull/1597/commits/3636ca2609034007496a09188f0076b786cf0803:
https://github.com/0xPolygon/polygon-edge/actions/runs/5230453010
- https://github.com/0xPolygon/polygon-edge/pull/1597/commits/8a2d3331c7d95a812a4f0022980b08a0bccb6ae6:
https://github.com/0xPolygon/polygon-edge/actions/runs/5230754692

File: state/txn.go (hunk 2 above)
Comment (Stefan-Ethernal) [reply state/txn.go:495]: Tried to fix it https://github.com/0xPolygon/polygon-edge/pull/1597/commits/81f4b5d709d3a736d47e29b6749c802e101b4123. If it turns out that it is not correct, we can revert it or re-work it.

File: state/immutable-trie/trie.go (hunk 1 above)
Comment (Stefan-Ethernal) [reply state/immutable-trie/trie.go]: Attempt to change it: https://github.com/0xPolygon/polygon-edge/pull/1597/commits/0fa21d9c5571b2a679d441b906211218abbb5cd1

**Review by igorcrevar — APPROVED** [review]

**Review by Stefan-Ethernal — APPROVED** [review]
LGTM

**Pull request merged**